src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -2368,19 +2224,63 @@ public boolean snapshotExists(String snapshotName)
      */
     public void clearSnapshot(String snapshotName)
     {
-        RateLimiter clearSnapshotRateLimiter = DatabaseDescriptor.getSnapshotRateLimiter();
-
-        List<File> snapshotDirs = getDirectories().getCFDirectories();
-        Directories.clearSnapshot(snapshotName, snapshotDirs, clearSnapshotRateLimiter);
+        SnapshotManager.instance.clearSnapshot(keyspace.getName(), getTableName(), snapshotName);
     }
+
     /**
      *
      * @return  Return a map of all snapshots to space being used
      * The pair for a snapshot has true size and size on disk.
      */
     public Map<String, TableSnapshot> listSnapshots()
     {
-        return getDirectories().listSnapshots();
+        Set<TableSnapshot> snapshots = new SnapshotLoader(getDirectories()).loadSnapshots();
+        Map<String, TableSnapshot> tagSnapshotsMap = new HashMap<>();
+
+        for (TableSnapshot snapshot : snapshots)
+            tagSnapshotsMap.put(snapshot.getTag(), snapshot);
+
+        return tagSnapshotsMap;
+    }
+
+    public Refs<SSTableReader> getSnapshotSSTableReaders(String tag) throws IOException
```

Comment:
Can this be moved to `TableSnapshot` ? I don't think `ColumnFamilyStore` needs to be involved in how snapshot sstables are read.

Reply:
I think this is just shuffling stuff around but if you insist ... putting that to TableSnapshot would need to be a static method.